### PR TITLE
add support for top-by-country and top-per-country pageview endpoints

### PR DESCRIPTION
**Pull request by geohci**
Issue #16

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,6 +17,8 @@ p = PageviewsClient(user_agent="<person@organization.org> Selfie, Cat, and Dog a
 p.article_views('en.wikipedia', ['Selfie', 'Cat', 'Dog'])
 p.project_views(['ro.wikipedia', 'de.wikipedia', 'commons.wikimedia'])
 p.top_articles('en.wikipedia', limit=10)
+p.top_articles_by_country('en.wikipedia', limit=10)
```
Comment (milimetric) [inline README.md:20]: should probably be `top_by_country`, as this returns a list of countries ordered from highest to lowest traffic on this wiki.  Sorry for the confusing naming of the endpoints, the one below should indeed have been `top_articles_per_country` and this one is really hard to name so maybe `top_by_country` is ok.

File: README.md [hunk 2]
```diff
@@ -17,6 +17,8 @@ p = PageviewsClient(user_agent="<person@organization.org> Selfie, Cat, and Dog a
...
+p.top_articles_per_country('in', limit=10)
```
Comment (milimetric) [inline README.md:21]: I leave this up to you, if you want to make it easier for search/replace later, you can make it `top_per_country` but I agree that `top_articles_per_country` is a much better name.

File: mwviews/api/pageviews.py [hunk 3]
```diff
@@ -301,6 +303,120 @@ def top_articles(
             'The pageview API returned nothing useful at: {}'.format(url)
         )
 
+    def top_articles_by_country(
```
Comment (milimetric) [inline mwviews/api/pageviews.py:306]: `top_by_country` here to match the above

File: mwviews/api/pageviews.py [hunk 4]
```diff
@@ -301,6 +303,120 @@ def top_articles(
...
+            self, project, access='all-access',
+            year=None, month=None, limit=1000):
+        """
+        Get pageviews to a project, split by country of origin for a given month
```
Comment (milimetric) [inline mwviews/api/pageviews.py:310]: Maybe to make it a little clearer:

Get a list of countries that access this project, sorted from highest to lowest pageview traffic, for a given month

File: mwviews/api/pageviews.py [hunk 5]
```diff
@@ -301,6 +303,120 @@ def top_articles(
...
+                    views_ceil: <int>
+                }
+                ...
+            ]
+        """
+        yesterday = date.today() - timedelta(days=1)
+        year = str(year or yesterday.year)
+        month = str(month or yesterday.month).rjust(2, '0')
+        day = str(day or yesterday.day).rjust(2, '0')
+        country = country.upper()  # common issue is lower-case country codes
```
Comment (milimetric) [inline mwviews/api/pageviews.py:400]: nice catch!

**Review by milimetric — COMMENTED** [review]
Looks great, just a naming suggestion really

File: mwviews/api/pageviews.py (hunk 4 above)
Comment (geohci) [reply mwviews/api/pageviews.py]: Great -- updated (though I took out "list of" so it wouldn't exceed Python's suggested line length)

File: mwviews/api/pageviews.py (hunk 5 above)
Comment (geohci) [reply mwviews/api/pageviews.py:400]: thanks! where "common issue" = I did it :)

File: mwviews/api/pageviews.py (hunk 3 above)
Comment (geohci) [reply mwviews/api/pageviews.py]: fixed

File: README.md (hunk 2 above)
Comment (geohci) [reply README.md]: agreed -- no perfect name and this aligns with the REST API naming so I switched it to `top_per_country`

File: README.md (hunk 1 above)
Comment (geohci) [reply README.md]: yep -- the dangers of copy-and-pasting! I updated to `top_by_country` to keep it simple and consistent with the API

**Comment by geohci** [comment]
Ok, updates made. thanks for the review! I'll let you merge but I also realize I don't know who handles package management on pip (or is that automatic?)

File: mwviews/api/pageviews.py (hunk 5 above)
Comment (milimetric) [reply mwviews/api/pageviews.py:400]: oh, me too :)

**Comment by milimetric** [comment]
thanks!  Looks great.  I don't know who does the pip thing either, I'll poke around

**Comment by milimetric** [comment]
Ok, figured it out.  It's me!  I'll make the changes necessary and publish.  I also moved the `master` branch to `main` 'cause that was bothering me.

**Comment by milimetric** [comment]
Ok, this was deployed as mwviews 0.2.0

**Pull request merged**